### PR TITLE
Add type mapping section to Redis connector

### PR DESCRIPTION
**Pull request by Jessie212**
## Non-technical explanation

Add type mapping section to Redis connector.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/redis.rst [hunk 1]
```diff
@@ -246,9 +246,68 @@ Field           Required  Type           Description
...
+------------
 
-In addition to the above Kafka types, the Redis connector supports ``hash`` type for the ``value`` field which represent data stored in the Redis hash.
+Because Trino and Redis each support types that the other does not, this
+connector :ref:`maps some types <type-mapping-overview>` when
+reading data.
+
+The following table lists Trino data types, which can be used in
+the ``type`` attribute, and the matching field decoders which can be specified in
+the ``dataFormat`` attribute:
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/redis.rst:258]: Remove duplicate copy

File: docs/src/main/sphinx/connector/redis.rst [hunk 2]
```diff
@@ -246,9 +246,68 @@ Field           Required  Type           Description
...
++-------------------------------------+--------------------------------------------------------------------------------+
+| | ``TIME WITH TIME ZONE``           | ``custom-date-time``, ``iso8601``                                              |
++-------------------------------------+--------------------------------------------------------------------------------+
+| | ``TIMESTAMP``                     | ``custom-date-time``, ``iso8601``, ``rfc2822``,                                |
+| |                                   | ``milliseconds-since-epoch``, ``seconds-since-epoch``                          |
++-------------------------------------+--------------------------------------------------------------------------------+
+| | ``TIMESTAMP WITH TIME ZONE``      | ``custom-date-time``, ``iso8601``, ``rfc2822``, ``milliseconds-since-epoch``   |
+| |                                   | ``seconds-since-epoch``                                                        |
++-------------------------------------+--------------------------------------------------------------------------------+
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/redis.rst:307]: Convert to standard `list-table` format

File: docs/src/main/sphinx/connector/redis.rst [hunk 3]
```diff
@@ -246,9 +246,68 @@ Field           Required  Type           Description
...
+  JSON object.
+* ``formatHint`` - Only for ``custom-date-time``.
+
+The JSON decoder supports multiple field decoders, with ``_default`` being
+used for standard table columns and a number of decoders for date- and
+time-based types.
+
+The following table lists Trino data types, which can be used as in ``type``,
+and matching field decoders, which can be specified via ``dataFormat``
+attribute.
```
Comment (jhlodin) [inline docs/src/main/sphinx/connector/redis.rst:281]: Suggest rewording for simplicity, something like:

> The following table lists Trino data types and their matching field decoders, which are specified in the `type` and `dataFormat` attributes respectively.

**Review by hashhar — COMMENTED** [review]
This is partial information. The Redis connector does support exactly all the decoders as mentioned in the Kafka connector docs.

Unless we want to duplicate the content in two places I'd say to leave this page as it is.

Maybe just add a `Type mapping` header under which the existing sentences can go and link to Kafka docs.

**Comment by jhlodin** [comment]
We should reduce "to learn about this connector, go read the docs for this other connector" type links as much as possible. Readers don't want to look at a doc for connector X when they're trying to configure connector Y. So, I do think we should duplicate the content into this page, possibly even as a fragment if they're identical. @Jessie212 @mosabua thoughts?

**Comment by hashhar** [comment]
Fragment sounds good.

Note that at the code level the decoders exist as a standalone thing so you can even have a separate page for them and link to them - the benefit being you can add more detailed examples per each decoder without having the doc page becoming excessively long. Also people generally only care about 1 of the possible options so looking at all of them on same page is probably not so useful.

**Comment by mosabua** [comment]
I think we should go down the route of using fragments since the same code applies. Thank you for that information @hashhar

**Comment by mosabua** [comment]
Any status update here @Jessie212 @jhlodin ?

**Comment by Jessie212** [comment]
Closing PR. Edits completed in separate PR: https://github.com/trinodb/trino/pull/17573.
